pkg/ebpf/c/tracee.bpf.c

```diff
@@ -4455,6 +4506,83 @@ int BPF_KPROBE(trace_security_inode_mknod)
     return events_perf_submit(&data, SECURITY_INODE_MKNOD, 0);
 }
 
+SEC("kprobe/do_splice")
+TRACE_ENT_FUNC(do_splice, DIRTY_PIPE_SPLICE);
+
+SEC("kretprobe/do_splice")
+int BPF_KPROBE(trace_ret_do_splice)
+{
+    args_t saved_args;
+    if (load_args(&saved_args, DIRTY_PIPE_SPLICE) != 0) {
+        // missed entry or not traced
+        return 0;
+    }
+
+    event_data_t data = {};
+    if (!init_event_data(&data, ctx))
+        return 0;
+
+    // dirty_pipe_splice is a splice to a pipe which results that the last page copied could be modified (the
+    // PIPE_BUF_CAN_MERGE flag is on in the pipe_buffer struct).
+    if (!event_chosen(DIRTY_PIPE_SPLICE))
+        return 0;
+
+    struct file *out_file = (struct file*)saved_args.args[2];
+    struct inode *out_inode = READ_KERN(out_file->f_inode);
+    struct pipe_inode_info *out_pipe = READ_KERN(out_inode->i_pipe);
+    // Check that output is a pipe
+    if (!out_pipe) {
+        return 0;
+    }
+
+    struct pipe_buffer *last_write_page_buffer = get_last_write_pipe_buffer(out_pipe);
+    unsigned int out_pipe_last_buffer_flags = READ_KERN(last_write_page_buffer->flags);
+    // This is not a dirty-pipe splice if CAN_MERGE flag is off.
+    if ((out_pipe_last_buffer_flags & PIPE_BUF_FLAG_CAN_MERGE) == 0) {
+        return 0;
+    }
+
+    u64 out_inode_number = READ_KERN(out_inode->i_ino);
+    unsigned int out_pipe_last_buffer_next_data_offset = READ_KERN(last_write_page_buffer->offset) + READ_KERN(last_write_page_buffer->len);
```

Comment:
I assume you're informing last buffer `offset + len` here so you know what data was exposed (or could have been) during the pipe buf merge attempt (dividing file contents / page_size and moving cursor to `offset + len`).

Reply:
I will think of a better name to describe it, so it won't be hard to understand why I calculate this value.